src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql.java

```diff
@@ -430,4 +454,80 @@ private teammates.storage.sqlentity.FeedbackResponseComment createFeedbackRespon
         return newComment;
     }
 
+    // methods for misc migration methods ---------------------------------------------------------------------------------
+
+    private void migrateInstructorEntities(teammates.storage.sqlentity.Course newCourse,
+            Map<String, User> userEmailToUserMap) {
+        List<Instructor> oldInstructors = ofy().load().type(Instructor.class).filter("courseId", newCourse.getId())
+                .list();
+        for (Instructor oldInstructor : oldInstructors) {
+            teammates.storage.sqlentity.Instructor newInstructor = migrateInstructor(newCourse, oldInstructor);
+            userEmailToUserMap.put(oldInstructor.getEmail(), newInstructor);
+        }
+    }
+
+    private teammates.storage.sqlentity.Instructor migrateInstructor(teammates.storage.sqlentity.Course newCourse,
+            Instructor oldInstructor) {
+        InstructorPrivileges newPrivileges;
+        if (oldInstructor.getInstructorPrivilegesAsText() == null) {
+            newPrivileges = new InstructorPrivileges(oldInstructor.getRole());
+        } else {
+            InstructorPrivilegesLegacy privilegesLegacy = JsonUtils
```

Comment:
Can use this constructor with `oldInstructor.getInstructorPrivilegesAsText`
https://github.com/TEAMMATES/teammates/blob/10e96cba2075d38da8843dc5611a41105c8b9941/src/main/java/teammates/common/datatransfer/InstructorPrivileges.java#L104

Reply:
Currently checking on this with others first, as the current logic is being used in `InstructorAttributes.java`, and is more likely to be correct than ` public InstructorPrivileges(String instrRole) `